src/append_only_data.rs

```diff
@@ -399,6 +390,18 @@ macro_rules! impl_appendable_data {
                     },
                 })
             }
+
+            pub fn check_permission(&self, request: Request, requester: PublicKey) -> Result<()> {
+                if unwrap!(self.inner.owners.last()).public_key == requester {
```

Comment:
Generally, we should not use `unwrap`s in the production code.
Given this function returns a `Result`, we can do something like this instead:
```suggestion
                if self.inner.owners.last().ok_or_else(|| Error::InvalidState)?.public_key == requester {
```